Add Strands Agents integration guide (#4607)
temporalio/documentation: docs/develop/python/integrations/strands-agents.mdx

---
id: strands-agents
title: Strands Agents integration
sidebar_label: Strands Agents
toc_max_heading_level: 3
keywords:
  - ai
  - agents
  - strands
  - strands agents
  - durable execution
  - ai workflows
tags:
  - Strands Agents
  - Python SDK
  - Temporal SDKs
description: Run Strands Agents AI workflows with durable execution using the Temporal Python SDK and Strands plugin.
---

> Temporal's integration with [Strands Agents](https://strandsagents.com/) is a [Plugin](/develop/plugins-guide) that

Author: ```suggestion
Temporal's integration with [Strands Agents](https://strandsagents.com/) is an [SDK Plugin](/develop/plugins-guide) that
```

gives your Strands agents [Durable Execution](/temporal#durable-execution) via the Temporal platform. The plugin routes

> model invocations, tool calls, MCP tool calls, and hooks through Temporal Activities, so every step the agent takes is

Author: ```suggestion
model invocations, tool calls, MCP tool calls, and hooks through Temporal Activities, so every step your agent takes is
```

recorded in Workflow history and can survive crashes, restarts, and infrastructure failures.

:::info

The Temporal Python SDK integration with Strands Agents is currently at an experimental release stage. The API may
change in future versions.

:::

Code snippets in this guide are taken from the
[Strands Agents plugin samples](https://github.com/temporalio/samples-python/tree/main/strands_plugin). Refer to the
samples for the complete code.

## Get started

Install the plugin, then run a minimal Strands agent inside a Temporal Workflow.

### Prerequisites

- This guide assumes you are already familiar with Strands Agents. If you are not, refer to the
  [Strands Agents documentation](https://strandsagents.com/) for more details.
- If you are new to Temporal, read [Understanding Temporal](/evaluate/understanding-temporal) or take the
  [Temporal 101](https://learn.temporal.io/courses/temporal_101/) course.
- Set up your local development environment by following the
  [Set up your local development environment](/develop/python/set-up-your-local-python) guide. Leave the Temporal
  development server running if you want to test your code locally.

### Install the plugin

> Install the Temporal Python SDK with Strands Agents support (requires `temporalio` 1.28.0 or later):

Author: TODO: Change this version number after release if needed

```

<!--SNIPEND-->

Register the Activity functions on the Worker:

<!--SNIPSTART python-strands-tools-worker-->

[strands_plugin/tools/run_worker.py](https://github.com/temporalio/samples-python/blob/main/strands_plugin/tools/run_worker.py)

```py
import asyncio
import os

from temporalio.client import Client
from temporalio.contrib.strands import StrandsPlugin
from temporalio.worker import Worker

from strands_plugin.tools.workflow import (
    ToolsWorkflow,
    environment_activity,
    fetch_weather,
)

async def main() -> None:
    plugin = StrandsPlugin()
    client = await Client.connect(
        os.environ.get("TEMPORAL_ADDRESS", "localhost:7233"),
        plugins=[plugin],
    )

    worker = Worker(
        client,
        task_queue="strands-tools",
        workflows=[ToolsWorkflow],
        activities=[fetch_weather, environment_activity],
    )
    print("Worker started. Ctrl+C to exit.")
    await worker.run()

if __name__ == "__main__":
    asyncio.run(main())
```

<!--SNIPEND-->

If you are using built-in `strands_tools`, wrap them in a thin async function decorated with `@activity.defn` so they
run as Temporal Activities.

### React to agent lifecycle events

> Strands' [hook system](https://strandsagents.com/) lets you subscribe callbacks to events in the agent lifecycle, such

Author: ```suggestion
Strands' [hook system](https://strandsagents.com/docs/user-guide/concepts/agents/hooks/) lets you subscribe callbacks to events in the agent lifecycle, such
```
